### ISSUE TITLE
Converting Tinker AMOEBA parameters with W-D-C Angle Terms

### ISSUE THREAD

**Issue by batukav** [openmm/openmm#3429]
Hello All,

We recently obtained a Tinker prm file for the POPC lipid that is based on AMOEBA force field and would like create a bilayer simulation using these parameters on OpenMM. 

My initial idea was to use processTinkerForceField.py (or built my own version based on this) for converting the prm file to an xml file. However, while going through the prm file I realised that the OPBENDTYPE is defined as the type W-D-C. It does not look like processTinkerForceField.py handles this term as the only supported OPBENDTYPE seems to be Allinger. 

I would like to ask if the WDC type for the OPBENDTYPE is currently supported by the OpenMM. If so, could anyone please make any suggestions how to include the WDC type conversion into the processTinkerForceField?

**Comment by peastman** [comment]
I'm not familiar with what that setting does.  I checked the Tinker manual but couldn't find any description of how the two types are different from each other.  Do you know what the difference is?

**Comment by batukav** [comment]
The most in-depth explanation I could find is here:
https://github.com/openmm/openmm/issues/2700#issuecomment-780107584

As I understand the W-D-C angle requires the projection of one of the atoms to the plane of the other three to define the out-of-plane-bending forces.

**Comment by peastman** [comment]
He's talking about `angle` and `anglep` terms.  The script supports both of them.  But that's different from `opbend`.

**Comment by batukav** [comment]
Indeed, I was lost more than I thought. 

I have looked into the Tinker9 source code (`seq_opbend.h`). The W-D-C angle is defined as  "angle between A-B-C plane and B-D vector for D-B<AC"  and the Allinger angle, which OpenMM uses, is defined as "angle between A-C-D plane and D-B vector for D-B<AC" . Other than that, I couldn't really figure out a difference in the energy functions for these angles.

Now I have realised that the AmoebaOutOfPlaneBendForce does not have the W-D-C implementation, as explained quite clearly in the manual. I should have read it more carefully before posting the question. Still, would it be possible to add this functionality to OpenMM (probably no time very soon but still)?

**Comment by peastman** [comment]
That shouldn't be too difficult.  The out of plane bend force is now implemented with a custom force:

https://github.com/openmm/openmm/blob/05bb471c5382717bce2a5f12b79178009332fc7f/wrappers/python/openmm/app/forcefield.py#L3726-L3734

We would just need to make it use a different expression depending on the value of that option.

I'm not sure how useful it would be though.  It looks to me like this is a global setting in Tinker?  So everything in your simulation has to use one form or the other.  You can't mix a lipid force field that uses one form with a protein force field that uses the other.  Is that correct?

**Comment by batukav** [comment]
> It looks to me like this is a global setting in Tinker? So everything in your simulation has to use one form or the other. You can't mix a lipid force field that uses one form with a protein force field that uses the other. Is that correct?

Yes, based on my limited understanding of Tinker, this is a global option that comes with the force field itself. With the particular force field I am trying to benchmark, because this OPBEND term is different, we won't be able to mix it with force fields that use the other term. 

All the other AMOEBA force fields I know are using Allinger definition as implemented in OpenMM. I can't say how useful it will be for the community but it might provide some sort of completeness.

**Comment by peastman** [comment]
Would it make sense to contact whoever created the POPC model to ask why they did it that way, and how it's intended to be used?  I don't object to adding this as a feature, but we should only do it if it will actually be useful.

**Comment by batukav** [comment]
Sure, I can ask for the details and the reasoning behind this. I understand your concern about this feature. I'll update here as soon as I learn more.
